docker/utils/decorators.py

```diff
@@ -35,3 +35,18 @@ def wrapper(self, *args, **kwargs):
             return f(self, *args, **kwargs)
         return wrapper
     return decorator
+
+
+def handle_interactive_response(f):
```

Comment:
Hm, since `_get_raw_response_socket` is used only in `attach_socket`, it doesn't make sense to use it anywhere else now. Not sure if it could be useable in future. Let me know if I should merge the functionality into `exec_start` directly or leave it as it is now.